### DISCUSSION TITLE
Node adapter: Preload all routes on start for faster initial load

### DISCUSSION THREAD

**Discussion by DePasqualeOrg** [sveltejs/kit#10033] [Q&A]
I'm running SvelteKit with the Node adapter, and I've noticed that the initial load of each route can be slow, especially when it uses heavy modules like Prisma. Is there a way to preload all of these routes when the app runs to speed up the initial load? So far the only solution I've been able to find is to load each of these routes with curl on the server just after running the app.

**Comment by teemingc** [comment]
You can either preload the code via https://kit.svelte.dev/docs/modules#$app-navigation-preloadcode
Or preload both code and data (from load functions) through https://kit.svelte.dev/docs/modules#$app-navigation-preloaddata

**Reply by DePasqualeOrg** [reply]
Thanks. Where would you call `preloadCode` in the app?

Another potential solution I thought of was to call `import '$lib/some/module';` in `hooks.server.ts`, but I'm not sure if this is a good idea.

**Reply by teemingc** [reply]
You can add it to a specific page or in a layout so it takes effect once for any pages under the layout.
I think it's something like this:
```svelte
<script>
	import { preloadCode } from "$app/navigation";
	import { onMount } from "svelte"

	onMount(() => {
		preloadCode(["/", "/about"]);
	})
</script>
```
Note that this method cannot be called on the server. This is why we call it after the page has mounted in the browser.

**Reply by DePasqualeOrg** [reply]
In that case, it looks like the code for those routes would load the first time someone visits a page under that layout, so the first page load would still be slow. I tried the solution I mentioned above with `hooks.server.ts`. It looks a bit like a hack, but it seems to work.

**Reply by teemingc** [reply]
Any idea what's causing the first page load to be slow? Is SSR enabled or disabled? Any large client-side / server-side libraries?

**Reply by DePasqualeOrg** [reply]
SSR is enabled. The issue is large libraries like Prisma being loaded on the first visit to a page with a server load function that uses them. Adding something like `import '$lib/server/prisma-client';` to `hooks.server.ts` causes these modules to load once the app starts.
